File: infopedia-user/src/main/java/com/iorix2k2/infopedia/user/model/Gender.java
package com.iorix2k2.infopedia.user.model;

import org.apache.commons.lang.StringUtils;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;


public enum Gender
{
	MALE("male"),
	FEMALE("female");
	
	private Gender(String text)
	{
		this.text = text;
	}
	
	@JsonValue
	public String getText()
	{
		return text;
	}
	
	@JsonCreator
	public static Gender createFromText(String text)
	{
		String t = StringUtils.defaultString(text).toLowerCase().trim();
		
		if(t.equals("male"))
			return Gender.MALE;
		else if(t.equals("female"))
			return Gender.FEMALE;
		
		return null;
	}
	
	
	private String text;
}

File: infopedia-frontend/src/main/webapp/WEB-INF/frontend/src/model/Gender.ts
export enum Gender {
  male = "male",
  female = "female"
}

class GenderUtil {
  public createFrom(text?: string) {
    const g = text?.toLowerCase();
    return g === "female" ? Gender.female : Gender.male;
  }

  public isValid(text?: string) {
    const t = text?.toLowerCase();
    return t === "male" || t === "female";
  }

  public getValues() {
    return [Gender.male, Gender.female];
  }
}



export default Gender;
export const genderUtil = new GenderUtil();